tools/env-loader/pkg/envloader.go

```diff
@@ -69,35 +74,59 @@ func findGitRepoRoot() (string, error) {
 	}
 }
 
-// Finds environment value files for the given environment and value set, under the given directory.
-// File names will be returned in order of priority, with the lowest priority names first.
-func FindEnvironmentFilesInDirectory(environmentsDirectoryPath, environmentName, valueSet string) ([]string, error) {
-	commonFiles, err := filepath.Glob(filepath.Join(environmentsDirectoryPath, CommonFileGlob))
-	if err != nil {
-		return nil, trace.Wrap(err, "failed to find common value files")
-	}
+// Given a base path (typically the environments directory) and a relative subdirectory path
+// (typically an environment name), find all common value files in every directory between
+// the two, inclusively.
+// NOTE: this assumes '/' is used a directory separator character. This is important so that
+// the same results are produced on multiple platforms.
+func findCommonFilesInPath(basePath, relativeSubdirectoryPath string) ([]string, error) {
+	relativeSubdirectoryPath = filepath.Clean(relativeSubdirectoryPath)
+	subdirectoryNames := strings.Split(relativeSubdirectoryPath, EnvironmentNameDirectorySeparator)
```

Comment:
Nit: use `path.Split` if you always want to split on `/`, or `filepath.Split` if you want to split on the OS-specific path separator.

Reply:
> path.Split
> filepath.Split

Unfortunately these just split the last item, rather than all of them (similar to `dirname`/`basename`). Instead, I want to split every item into a list. That is, convert `"a/b/c/d"` into `[]string{"a", "b", "c", "d"}`.

Is there a function that does this? I checked both `path` and `filepath` and didn't see anything, but I might have missed something.

Reply:
Oof, I see.

Then no I think `strings.Split` is the best you can do.

If you are going to continue to split on `/` then it wouldn't hurt to do `filepath.ToSlash()` on the input on the off chance this code ever runs on a platform like Windows.

Reply:
> on the off chance this code ever runs on a platform like Windows.

This isn't an off chance - it's a certainty. This will run on every platform that we build on. That said this is a good idea - added.

Reply:
Ok that works for me, but I think you want to invert the order. Do the `filepath.Clean` first (because it's going to operate on the OS-specific separator), and then send convert the cleaned path to a uniform separator with `filepath.ToSlash`.

The other option, which seems a bit better now that I think about it, would be to eliminate the `ToSlash` and instead do a `strings.Split(s, filepath.Separator)` and get rid of your `EnvironmentNameDirectorySeparator` constant.

Reply:
> strings.Split(s, filepath.Separator)

I considered this, but don't think it's what we want here. When we use this tool in our build pipelines, we will do something like:

`env-loader --environment "prod/publish" --value-set gpg-signing >> "${GITHUB_ENV}"`

Ultimately the separator character is one chosen for convenience (it's the separator used in both URLs and unix system file systems), and strictly speaking isn't expected to be the OS file path separator character. IMO it would be better to keep the environment name consistent across all platforms, rather than slightly different for non-unix platforms. This should help prevent bugs when it's used in practice.